### PR TITLE
A test configuration for the environment based on local shell scripts

### PR DESCRIPTION
**Pull request by motus**
# Pull Request

## Title

A test configuration for the environment based on local shell scripts

---

## Description

This is a sample MLOS configuration (CLI config, tunables, environment, and the scripts) that launches a local environment and invokes local shell scripts to set it up and run trials.

---

## Type of Change

This is purely a config-based update. We might add unit tests around that setup later.

- ✨ New feature
- 📝 Documentation update
- 🧪 Tests

---

## Testing

Unit tests that validate the environment and run test experiments are included in this PR.

To test the environment manually, run:
```bash
mlos_bench \
     --config mlos_bench/mlos_bench/tests/config/cli/test-cli-local-env-bench.jsonc \
     --globals experiment_test_local.jsonc \
     --tunable_values tunable-values/tunable-values-local.jsonc
```

---

## Additional Notes (optional)

This setup is supposed to serve as an example of other local environments with shell scripts.

---

### CODE REVIEW

File: mlos_bench/mlos_bench/tests/config/environments/local/scripts/bench_run.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+#
+"""
+Helper script to run the benchmark and store
+the results and telemetry in CSV files.
+
+Run:
+    ./bench_run.py ./output-metrics.csv ./output-telemetry.csv`
+"""
```
Comment (bpkroth) [inline mlos_bench/mlos_bench/tests/config/environments/local/scripts/bench_run.py:12]: I think you need to run `make format` and checkin the changes for the CI to pass.

File: mlos_bench/mlos_bench/tests/config/cli/test-cli-local-env-bench.jsonc
Comment (bpkroth) [inline mlos_bench/mlos_bench/tests/config/cli/test-cli-local-env-bench.jsonc:1]: Would be good to add this to a test that actually runs in the CI as well so we know that the example actually works :)

File: mlos_bench/mlos_bench/tests/config/cli/test-cli-local-env-opt.jsonc
Comment (bpkroth) [inline mlos_bench/mlos_bench/tests/config/cli/test-cli-local-env-opt.jsonc:1]: Same here :)

File: mlos_bench/mlos_bench/tests/config/cli/test-cli-local-env-bench.jsonc
Comment (motus) [reply mlos_bench/mlos_bench/tests/config/cli/test-cli-local-env-bench.jsonc:1]: yes, that is in the plans

File: mlos_bench/mlos_bench/tests/config/environments/local/scripts/bench_run.py (hunk 1 above)
Comment (motus) [reply mlos_bench/mlos_bench/tests/config/environments/local/scripts/bench_run.py:11]: fixed!

File: mlos_bench/mlos_bench/tests/config/cli/test-cli-local-env-bench.jsonc
Comment (motus) [reply mlos_bench/mlos_bench/tests/config/cli/test-cli-local-env-bench.jsonc:1]: done! I've added both -opt and -bench CLIs to the unit tests

File: mlos_bench/mlos_bench/tests/config/environments/local/scripts/bench_run.py
Comment (bpkroth) [inline mlos_bench/mlos_bench/tests/config/environments/local/scripts/bench_run.py:1]: Might be good to put some more comments in this script to very explicitly note that this is a dummy and meant to take the place of a real script that runs a workload and extracts the results to be output in csv format for the optimizer and storage layer in mlos_bench to receive.
That way people looking at the example will hopefully have a more clear idea of what to change.

Comment (bpkroth) [reply mlos_bench/mlos_bench/tests/config/environments/local/scripts/bench_run.py:1]: Might also be good for us to put that into the template repo to make it easier to see and adapt outside of the core MLOS repo.

**Review by bpkroth — APPROVED** [review]

**Comment by bpkroth** [comment]
Lgtm. A few extra explanation comments wouldn't hurt, but can expand those in a future PR too.

File: mlos_bench/mlos_bench/tests/config/environments/local/scripts/bench_run.py
Comment (motus) [reply mlos_bench/mlos_bench/tests/config/environments/local/scripts/bench_run.py:1]: makes sense. will do it in a moment

Comment (motus) [reply mlos_bench/mlos_bench/tests/config/environments/local/scripts/bench_run.py:1]: done

**Pull request merged**